MUXSDKStats/MUXSDKStats/MUXSDKPlayerBinding.m

```diff
@@ -400,13 +399,26 @@ - (void)checkVideoData {
     if ([currentPlayerAsset isKindOfClass:AVURLAsset.class]) {
         AVURLAsset *urlAsset = (AVURLAsset *)currentPlayerAsset;
         NSString * urlString = [[urlAsset URL] absoluteString];
-        if (!_videoURL || [_videoURL isEqualToString:urlString]) {
+        if (!_videoURL || ![_videoURL isEqualToString:urlString]) {
             _videoURL = urlString;
             videoDataUpdated = YES;
         }
     }
-    if (_lastAdvertisedBitrate > 0) {
+    if (![@(_lastDispatchedAdvertisedBitrate) doubleValueIsEqual:@(_lastAdvertisedBitrate)]) {
         videoDataUpdated = YES;
+        _lastDispatchedAdvertisedBitrate = _lastAdvertisedBitrate;
+        _sourceDimensionsHaveChanged = YES;
+    }
+    if (_sourceDimensionsHaveChanged && CGSizeEqualToSize(_videoSize, _lastDispatchedVideoSize)) {
+        CGSize sourceDimensions = [self getSourceDimensions];
+        if (!CGSizeEqualToSize(_videoSize, sourceDimensions)) {
+            _videoSize = sourceDimensions;
+            if (sourceDimensions.width > 0 && sourceDimensions.height > 0) {
+                _lastDispatchedVideoSize = _videoSize;
+                _sourceDimensionsHaveChanged = NO;
+                videoDataUpdated = YES;
+            }
+        }
```

Comment:
Now the real magic, if source dimensions have changed then we compare _videoSize to sourceDimensions if they are not equal then we know `sourceDimensions` has changed.

Next, we make sure the `sourceDimensions` value is valid.

And if so then we reset `_sourceDimensionsHaveChanged`

----

The above block will happen repeatedly (every ~0.1s) until we get valid values for `sourceDimensions`, correct? And the reason for that is that we don't know when that value is populated. In AVPlayer world that value is "eventually correct".

That's unfortunate but we have to work with it.

Reply:
>The above block will happen repeatedly (every ~0.1s) until we get valid values for sourceDimensions, correct? 

That is correct